compiler/infra/OMRMonitor.hpp

```diff
@@ -47,18 +47,23 @@ class Monitor
    static TR::Monitor *create(char *name);
    static void destroy(TR::Monitor *monitor);
    void enter();
-   int32_t try_enter() { TR_UNIMPLEMENTED(); }
+   int32_t try_enter() { TR_UNIMPLEMENTED(); return 0; }
    int32_t exit(); // returns 0 on success
    void destroy();
    void wait() { TR_UNIMPLEMENTED(); }
-   intptr_t wait_timed(int64_t millis, int32_t nanos) { TR_UNIMPLEMENTED(); }
+   intptr_t wait_timed(int64_t millis, int32_t nanos) { TR_UNIMPLEMENTED(); return static_cast<intptr_t>(0); }
```

Comment:
Why is a `static_cast` needed to `intptr_t`, but not `int32_t` (above)?

Reply:
I've removed commit dc392586d which included these `return` statements as #3610 has properly fixed `TR_UNIMPLEMENTED` to mark it with `OMR_NORETURN`. These changes no longer exist.